### PR TITLE
Removes web3, fixes some dependencies that openzeppelin test-helpers has on web3

### PR DESCRIPTION
**Pull request by xklob**

### CODE REVIEW

**Comment by ditchfieldcaleb** [comment]
Also moves old scripts & dao proposals into an "old" folder within their respective parent folders, so that typescript doesn't try to compile them.

**Review by Joeysantoro — COMMENTED** [review]
.

File: scripts/utils/helpers.ts [hunk 1]
```diff
@@ -13,10 +13,12 @@ export function check(flag, message) {
 
 // The current version of the contracts in the repo uses readOracle as the api
 // The old api was peg(), so the currently deployed contracts need manual calling until the upgrade
+/*
```
Comment (Joeysantoro) [inline scripts/utils/helpers.ts:16]: this function can be deleted

File: test/helpers.ts [hunk 2]
```diff
@@ -134,13 +132,79 @@ async function expectApprox(actual, expected, magnitude = '1000') {
...
+  advanceBlockTo: async (target: number | string | BigNumberish) => {
+    target = ethers.BigNumber.from(target);
+
+    const currentBlock = await time.latestBlock();
+    const start = Date.now();
+    let notified;
+    if (target.lt(currentBlock))
+      throw Error(`Target block #(${target}) is lower than current block #(${currentBlock})`);
+    while (ethers.BigNumber.from(await time.latestBlock()).lt(target)) {
+      if (!notified && Date.now() - start >= 5000) {
```
Comment (Joeysantoro) [inline test/helpers.ts:187]: incredible attention to detail

File: test/integration/e2e.spec.ts [hunk 3]
```diff
@@ -1518,17 +1425,25 @@ describe('e2e', function () {
...
       it('should be able to properly set rewards on the rewards distributor', async function () {
         const { rariRewardsDistributorDelegator, rariPool8Tribe } = contractAddresses;
-        const seventyFiveTribe = toBN('75').mul(toBN(e18));
+        const tribalChief = contracts.tribalChief as TribalChief;
+
+        const elevenTribe = ethers.constants.WeiPerEther.mul('11');
+        const tribeReward = await tribalChief.tribePerBlock();
+
+        const contractTx = await tribalChief.updateBlockReward(elevenTribe);
+        await contractTx.wait();
```
Comment (Joeysantoro) [inline test/integration/e2e.spec.ts:1434]: can this wait be removed?

File: test/integration/e2e.spec.ts [hunk 4]
```diff
@@ -1518,17 +1425,25 @@ describe('e2e', function () {
...
         );
 
-        const expectedNewCompSpeed = seventyFiveTribe.mul(toBN(poolAllocPoints)).div(toBN(totalAllocPoint));
+        const expectedNewCompSpeed = elevenTribe.mul(`${poolAllocPoints}`).div(totalAllocPoint);
         const [newCompSpeed, updateNeeded] = await autoRewardsDistributor.getNewRewardSpeed();
-        expect(newCompSpeed).to.be.equal(expectedNewCompSpeed);
+        expect(toBN(newCompSpeed)).to.be.equal(expectedNewCompSpeed);
         expect(updateNeeded).to.be.true;
 
+        console.log('Setting auto rewards distribution');
```
Comment (Joeysantoro) [inline test/integration/e2e.spec.ts:1446]: remove console log

File: test/integration/e2e.spec.ts [hunk 5]
```diff
@@ -1601,4 +1519,87 @@ describe('e2e', function () {
...
+      }
+
+      doLogging && console.log(`Testing governor role...`);
+      for (let i = 0; i < accessControl.governor.length; i += 1) {
+        const contractAddress = accessControl.governor[i];
+        const isGovernor = await core.isGovernor(contractAddress);
+        expect(isGovernor).to.be.equal(true);
+      }
+
+      /*
```
Comment (Joeysantoro) [inline test/integration/e2e.spec.ts:1597]: Make a ticket for this to not forget

File: test/helpers.ts (hunk 2 above)
Comment (ditchfieldcaleb) [reply test/helpers.ts:187]: Definitely made this myself with no reference ;)

**Comment by ditchfieldcaleb** [comment]
@Joeysantoro can you fix the merge conflicts? Gotta head to bed. Fixed the other issues though.

File: contract-addresses/mainnetAddresses.ts [hunk 6]
```diff
@@ -97,9 +97,9 @@ const MainnetAddresses = {
     artifactName: 'ERC20CompoundPCVDeposit',
     address: '0x243C601CC5DaA3Ac250B14509804188347bd2aFB'
   },
-  curve3crv: { artifactName: 'IStableSwap3', address: '0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490' },
-  curve3pool: { artifactName: 'IStableSwap3', address: '0xbebc44782c7db0a1a60cb6fe97d0b483032ff1c7' },
-  curveMetapool: { artifactName: 'IStableSwap2', address: '0x06cb22615ba53e60d67bf6c341a0fd5e718e1655' },
+  curve3crv: { artifactName: 'unknown', address: '0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490' },
+  curve3pool: { artifactName: 'unknown', address: '0xbebc44782c7db0a1a60cb6fe97d0b483032ff1c7' },
+  curveMetapool: { artifactName: 'unknown', address: '0x06cb22615ba53e60d67bf6c341a0fd5e718e1655' },
```
Comment (ElliotFriedman) [inline contract-addresses/mainnetAddresses.ts:102]: Why change the variable names and artifacts?

Comment (ditchfieldcaleb) [reply contract-addresses/mainnetAddresses.ts:102]: @ElliotFriedman the "IStableSwap" contracts aren't in the repo.
When the loader tries to grab the artifact and throw this into the mainnetContracts, it throws an error, causing an integration test to fail.

**Review by Joeysantoro — APPROVED** [review]

**Pull request merged**